DotNetDBF/DBFReader.cs

```diff
@@ -231,6 +242,38 @@ public object[] NextRecord()
             return NextRecord(_selectFields, _orderedSelectFields);
         }
 
+        /// <summary>
+        /// Reads the data table from DBF from current position to end.
+        /// </summary>
+        /// <returns>The data table with all data, original types and names of columns.</returns>
+        /// <param name="dataTableName">Name for data table.</param>
+        public DataTable ReadDataTable(string dataTableName)
+        {
+            if (_selectFields.Length == 0)
+                SelectAllFields();
+            DataTable DT = new DataTable(dataTableName);
+            foreach (int i in _orderedSelectFields)
+            {
+                DBFField DF = _header.FieldArray[i];
+                string n = DF.Name;
+                DT.Columns.Add(n, DF.Type);
+                if (DF.Type == typeof(System.String))
+                    DT.Columns[n].MaxLength = DF.FieldLength;
+            }

```

Comment:
I think you just merely needed to use GetSelectFields() instead of using the field array directly

Reply:
Good idea, thanks! https://github.com/ekonbenefits/dotnetdbf/pull/38/commits/bbe4e690f7c32d795d1e9439353076f9c39c8907